### PR TITLE
[Validator] Add the type for ConstraintValidatorTestCase::createValidator

### PR DESCRIPTION
**Pull request by stof**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no 
| Tickets       | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Review by chalasr — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Review by fancyweb — APPROVED** [review]

**Comment by xabbuh** [comment]
Do we need to add the changed file in the list of excluded classes in `patch-types.php`?

**Comment by stof** [comment]
indeed, we should (and ideally, we should update the patcher to account for `@template`)

**Comment by stof** [comment]
Looking at it, the patcher handles template type only when the method has a native return type.

Maybe this should be typed like that for now, to account for adding a native return type in Symfony 7.0:

```php
    /**
     * @return ConstraintValidatorInterface
     * @psalm-return T
     */
    abstract protected function createValidator();
```

This way, in 7.0, we can add `ConstraintValidatorInterface` as a native return type (and switch to `@return T` for the template).

**Comment by nicolas-grekas** [comment]
@stof that'd make sense to me.

**Comment by stof** [comment]
I updated the PR

**Review by yceruto — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
rebase needed

**Comment by stof** [comment]
@nicolas-grekas The rebase is done

File: src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php [hunk 1]
```diff
@@ -131,7 +132,7 @@ protected function createEntityManagerMock($repositoryMock)
         return $em;
     }
 
-    protected function createValidator()
+    protected function createValidator(): ConstraintValidatorInterface
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Tests/Validator/Constraints/UniqueEntityValidatorTest.php]:
```suggestion
    protected function createValidator(): UniqueEntityValidator
```

Covariance ftw?

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @stof.

**Pull request merged**